indexes/social.js

```diff
@@ -102,54 +103,61 @@ module.exports = function (log, dir) {
   const name = 'social'
   const { level, seq } = Plugin(log, dir, name, 1, handleData, writeData)
 
+  function getResults(opts, live, cb) {
+    pull(
+      pl.read(level, opts),
+      pull.collect((err, data) => {
+        if (err) return cb(err)
+        if (live) {
+          const ps = pull(
+            pl.read(level, Object.assign({}, opts, { live, old: false })),
+            pull.map(parseInt10)
+          )
+          cb(null, liveOffsets(data.map(parseInt10), ps))
+        } else cb(null, offsets(data.map(parseInt10)))
+      })
+    )
+  }
```

Comment:
Thought: I wonder if this could use `pull-cat` instead of being a nested callback looking thing.

Related to that, I wonder if `LIVEOFFSETS` could *only* have `ps`, and then we could do `and(offsets(data.map(parseInt10)), liveOffsets(ps))` and update the logic in JITDB so that LIVEOFFSETS is only about the pull-stream.

Just wondering of a refactor. This obviously only affects code style/quality, but arguably it's important so we don't end up with flumedb type of code quality.

Reply:
I was thinking about that when doing the jitdb PR, it requires a bit finesse around liveOffsets for the normal queries, but could be done I guess.